sql/ha_partition.cc

```diff
@@ -5663,6 +5663,19 @@ int ha_partition::read_range_next()
   DBUG_RETURN(handle_unordered_next(table->record[0], eq_range));
 }
 
+bool ha_partition::has_gap_locks()
```

Comment:
Originally I was going to add it just as a handlerton flag, then implement the virtual similar to has_transactions that just returns flagset & flag, but the handlerton flags seems to be getting full and is not possible/practical for the partition handler to accurately reflect what the underlying handlerton supports via the HTON flags. Also as mentioned elsewhere, it is my understanding (perhaps incorrect) that there are some ways to alter the engine of an individual partition within a partitioned table, thus requiring something like this to scan all of the partitions to ensure 100% support.